src/main/java/com/github/wovnio/wovnjava/CustomDomainLanguage.java

```diff
@@ -0,0 +1,39 @@
+package com.github.wovnio.wovnjava;
+
+import java.net.URL;
+import java.util.Arrays;
+
+class CustomDomainLanguage {
+    public final String host;
+    public final String path;
+    public final Lang lang;
+
+    public CustomDomainLanguage(String host, String path, Lang lang) {
+        this.host = host;
+        this.path = path;
+        this.lang = lang;
+    }
+
+    public boolean isMatch(URL url) {
+        if (url == null) return false;
+
+        return this.host.equalsIgnoreCase(url.getHost()) && this.pathIsEqualOrSubsetOf(this.path, url.getPath());
+    }
+
+    private boolean pathIsEqualOrSubsetOf(String basePath, String testPath) {
+        String[] basePathList = Arrays.stream(basePath.split("/")).filter(s -> !s.isEmpty()).toArray(String[]::new);
```

Comment:
I noticed you used the Stream API here but not other places, for example CustomDomainLanguages etc that are basic for loops with filtering.
This is also somewhat related to the Java 6/backwards compatibility discussion though, as streams are a Java 8 feature.

Reply:
Here I simply need to filter out empty elements of the array before doing further calculations, and this seemed to be the most concise way to do it.

I did not consider pre-java 8 support for these changes. But nice catch, we would have to change this filtering method for java 7 support. I'm torn on whether or not that means we should also change it for the java 8+ master branch.

Reply:
Definitely on board with using newer language features. Can address back-porting to older versions when it is needed.